listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -432,6 +479,64 @@ def get_recording_recommendation_events(users_for_events: List[dict], min_ts: in
     return events
 
 
+def get_cb_review_events(users_for_events: List[dict], min_ts: int, max_ts: int, count: int) -> List[APITimelineEvent]:
+    """ Gets all CritiqueBrainz review events in the feed.
+    """
+    id_username_map = {user['id']: user['musicbrainz_id'] for user in users_for_events}
+    cb_review_events_db = db_user_timeline_event.get_cb_review_events(
+        user_ids=[user['id'] for user in users_for_events],
+        min_ts=min_ts,
+        max_ts=max_ts,
+        count=count,
+    )
+
+    review_ids, entity_ids, review_id_event_map = [], [], {}
+    for event in cb_review_events_db:
+        review_id = event.metadata.review_id
+        review_ids.append(review_id)
+        review_id_event_map[review_id] = event
+        entity_ids.append(event.metadata.entity_id)
+
+    reviews = CritiqueBrainzService().fetch_reviews(review_ids)
+    if reviews is None:
+        return []
+
+    mbid_metadatas = fetch_mapped_recording_data(entity_ids)
```

Comment:
This works for recording reviews. But won't work for artist reviews or release group reviews. One way I think other reviews could be made playable is to always store recording mbid in the metadata even if it is a artist or release group review and play that. Thoughts?

Reply:
Also, we don't store any other metadata like artist, track name etc for reviews in LB currently and CB doesn't store that either. It also only stores the mbid and retrieves the data from MB as needed.